### ISSUE TITLE
Issue: Unable to Access Igotify App on iPhone 11

### ISSUE THREAD

**Issue by BuckerooBanza1** [androidseb25/iGotify-Notification-Assistent#105]
Description: When attempting to log in to the Igotify app on my iPhone 11, I receive the following error message: "You need to provide a valid access token or user credentials to access this API." I have verified my credentials multiple times and confirmed they are correct. I am also using the latest version of the app.

Additional Info:

An API call via the Swagger UI works with all tokens, but the Igotify app fails to recognize them.

The error persists despite all verification steps.

Can someone help me resolve this issue?

**Comment by BuckerooBanza1** [comment]
Yes, because i used the same config and app settings before and it worked just fine.
But i had too reinstall the app on my iphone and now i just isnt working anymore.

**Comment by BuckerooBanza1** [comment]
The versions are all correct

**Comment by BuckerooBanza1** [comment]
Edit:
It connects to my server and catches all the versions but when i enter my credentials the error above is showing up.

**Comment by androidseb25** [comment]
Ok mmmh what does the log in gotify container say when you try to login.

Did you have any special character in you're password.

I had in the past a bug that with some special character gotify want let login user via the api endpoint 🤔

**Comment by BuckerooBanza1** [comment]
So I have now made a completely new VM under Proxmox and used only letters and numbers as a password, it still doesn't work, here is the docker log:


gotify     | Starting Gotify version 2.5.0@2024-06-23-17:12:59
igotify    | Database is created: True
igotify    | Gotify Url list is: empty
igotify    | Gotify Client list is: empty
igotify    | SecNtfy Token list is: empty
igotify    | If one or more lists are empty please check the environment variable! GOTIFY_SERVERS or GOTIFY_CLIENTS or NTFY_TOKENS
igotify    | If all lists are empty do nothing, you will configure the gotify server over the iGotify app.
igotify    | info: Microsoft.Hosting.Lifetime[14]
igotify    |       Now listening on: http://[::]:8080
igotify    | info: Microsoft.Hosting.Lifetime[0]
igotify    |       Application started. Press Ctrl+C to shut down.
igotify    | info: Microsoft.Hosting.Lifetime[0]
igotify    |       Hosting environment: Production
igotify    | info: Microsoft.Hosting.Lifetime[0]
igotify    |       Content root path: /app
gotify     | Started listening for plain connection on tcp [::]:80
gotify     | 2024-10-16T15:43:52Z | 200 |    3.972148ms | 192.168.178.106 | GET      "/"
gotify     | 2024-10-16T15:43:52Z | 200 |    1.409118ms | 192.168.178.106 | GET      "/static/css/2.0f3898ba.chunk.css"
gotify     | 2024-10-16T15:43:52Z | 200 |    4.043402ms | 192.168.178.106 | GET      "/static/js/main.c9d23f31.chunk.js"
gotify     | 2024-10-16T15:43:52Z | 200 |    41.76522ms | 192.168.178.106 | GET      "/static/js/2.b786495f.chunk.js"
gotify     | 2024-10-16T15:43:53Z | 401 |     287.009µs | 192.168.178.106 | GET      "/current/user"
gotify     | Error #01: you need to provide a valid access token or user credentials to access this api
gotify     | 2024-10-16T15:43:53Z | 200 |     603.971µs | 192.168.178.106 | GET      "/static/media/roboto-latin-400.176f8f5b.woff2"
gotify     | 2024-10-16T15:43:53Z | 200 |      796.78µs | 192.168.178.106 | GET      "/static/media/roboto-latin-500.f5b74d7f.woff2"
gotify     | 2024-10-16T15:43:53Z | 401 |      222.02µs | 192.168.178.106 | GET      "/client"
gotify     | Error #01: you need to provide a valid access token or user credentials to access this api
gotify     | 2024-10-16T15:43:53Z | 401 |     125.116µs | 192.168.178.106 | GET      "/current/user"
gotify     | Error #01: you need to provide a valid access token or user credentials to access this api
gotify     | 2024-10-16T15:43:53Z | 401 |       21.75µs | 192.168.178.106 | GET      "/client"
gotify     | Error #01: you need to provide a valid access token or user credentials to access this api
gotify     | 2024-10-16T15:43:53Z | 200 |     1.69692ms | 192.168.178.106 | GET      "/static/favicon.ico"
gotify     | 2024-10-16T15:43:56Z | 200 |   130.89517ms | 192.168.178.106 | POST     "/client"
gotify     | 2024-10-16T15:43:56Z | 200 |   38.678012ms | 192.168.178.106 | GET      "/current/user"
gotify     | 2024-10-16T15:43:56Z | 200 |     226.284µs | 192.168.178.106 | GET      "/stream?token=[masked]"
gotify     | 2024-10-16T15:43:56Z | 200 |      273.46µs | 192.168.178.106 | GET      "/application"
gotify     | 2024-10-16T15:43:56Z | 200 |     332.212µs | 192.168.178.106 | GET      "/message?since=0"
gotify     | 2024-10-16T15:44:01Z | 200 |      313.45µs | 192.168.178.106 | GET      "/application"
gotify     | 2024-10-16T15:44:10Z | 200 |   68.692133ms | 192.168.178.106 | POST     "/application"
gotify     | 2024-10-16T15:44:10Z | 200 |     283.973µs | 192.168.178.106 | GET      "/application"
gotify     | 2024-10-16T15:44:10Z | 200 |     129.878µs | 192.168.178.106 | GET      "/static/defaultapp.png"
igotify    | warn: Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware[3]
igotify    |       Failed to determine the https port for redirect.
igotify    |
gotify     | 2024-10-16T15:45:59Z | 401 |      53.677µs |  192.168.178.44 | GET      "/client"
gotify     | Error #01: you need to provide a valid access token or user credentials to access this api

**Comment by BuckerooBanza1** [comment]
So I found out that it's a problem with my cloudflare tunnel reverse proxy. If I log in via the IP in my local network, it works without any problems. But if I use the domain that points to my IP and port 8680, I am redirected to the Login UI, but I can't log in and get the error You need to provide a valid access token or user credentials to access this API.

**Comment by BuckerooBanza1** [comment]
Ok I figured out how to work around the error: Only the login in the igotify app does not work via Cloudflare Tunnels. (Strangely enough, it works in the browser without any problems) That's why I have now logged into my local network in the I Gotify app via IP address of my Gotify server, which worked. Then I changed the IP to the domain and the port to 80 in the settings. This works without any problems. Unfortunately, you have to do this after every restart of the gotify server...

(In my case the tunnel and the gotify server is deployed on the same server via docker)

**Comment by androidseb25** [comment]
it's a cloudflare specific bug, still under investigation

**Comment by androidseb25** [comment]
did'nt find anything why tis error appears, i close the issue now. It can be reopen by any people who has the same issue

**Comment by androidseb25** [comment]
for upcoming update you will have the option to add custom headers to by pass maybe the Cloudflare problem with header values

the progress for the new Update you will find here #143
